core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -41,306 +40,337 @@ object ConsumerGroupCommand {
-          consumerOpt match {
-            case Some(consumer) =>
-              val topicAndPartition: TopicAndPartition = new TopicAndPartition(topicPartition.topic(), topicPartition.partition())
-              val request =
-                OffsetRequest(Map(topicAndPartition -> PartitionOffsetRequestInfo(OffsetRequest.LatestTime, 1)))
-              val logEndOffset = consumer.getOffsetsBefore(request).partitionErrorAndOffsets(topicAndPartition).offsets.head
-              consumer.close()
-              print(logEndOffset)
-            case None => // ignore
-          }
-        }
-      case None =>
-        println("No broker for partition %s".format(topicPartition))
+
+    def close() {
+      adminClient.close()
+      consumer.close()
     }
-  }
 
-  private def getNewConsumer(zkUtils: ZkUtils, brokerId: Int): Option[KafkaConsumer[String, String]] = {
-    try {
-      zkUtils.readDataMaybeNull(ZkUtils.BrokerIdsPath + "/" + brokerId)._1 match {
-        case Some(brokerInfoString) =>
-          Json.parseFull(brokerInfoString) match {
-            case Some(m) =>
-              val brokerInfo = m.asInstanceOf[Map[String, Any]]
-              val host = brokerInfo.get("host").get.asInstanceOf[String]
-              val port = brokerInfo.get("port").get.asInstanceOf[Int]
-              val deserializer: String = (new StringDeserializer).getClass.getName
-              val properties: Properties = new Properties()
-              properties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, host + ":" + port)
-              properties.put(ConsumerConfig.GROUP_ID_CONFIG, "ConsumerGroupCommand")
-              properties.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false")
-              properties.put(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, "30000")
-              properties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, deserializer)
-              properties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, deserializer)
-              Some(new KafkaConsumer[String, String](properties))
-            case None =>
-              throw new BrokerNotAvailableException("Broker id %d does not exist".format(brokerId))
-          }
-        case None =>
-          throw new BrokerNotAvailableException("Broker id %d does not exist".format(brokerId))
-      }
-    } catch {
-      case t: Throwable =>
-        println("Could not parse broker info due to " + t.getMessage)
-        None
+    private def createAdminClient(): AdminClient =
+      AdminClient.createSimplePlaintext(opts.options.valueOf(opts.bootstrapServerOpt))
+
+    private def createNewConsumer(): KafkaConsumer[String, String] = {
+      val properties = new Properties()
+      val deserializer = (new StringDeserializer).getClass.getName
+      val brokerUrl = opts.options.valueOf(opts.bootstrapServerOpt)
+      properties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerUrl)
+      properties.put(ConsumerConfig.GROUP_ID_CONFIG, opts.options.valueOf(opts.groupOpt))
```

Comment:
We probably need to use the previous default groupId when we're listing groups since no groupId would be provided in the options. Even better would be to not create a KafkaConsumer instance at all in that case.